[ie/YoutubeSearchDate] Fix search parameters

alb opened the pull request:
### Fix search parameters for YouTube extractor 

This PR fixes the search parameters for `YouTubeSearchDateIE` to use the following filters: `sort:date`, `filter:videos`.
I didn't update the parameters for `YouTubeSearchIE` as the current parameters still seem to be correct and testing on the website gives me the same parameters when applying the`filter:videos` filter manually.

Fixes #9156 

<details open>
<summary>Template</summary>

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

pukkandan commented:
SearchIE has the same issue. Pls update it too. I've mentioned the desired param in #9156. Also, add a test - perhaps the example in the issue

alb commented:
@pukkandan I've updated the search params for SearchIE as well now and added the example from the issue as a test along with a note as to why it's needed.

bashonly commented on yt_dlp/extractor/youtube.py:
```diff
@@ -6965,22 +6965,29 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
         }
+    }, {
+        'note': 'Suicide/self-harm search warning',
+        'url': 'ytsearch1:i hate myself and i wanna die',
+        'info_dict': {
```
```suggestion
        'url': 'ytsearch1:i hate myself and i wanna die',
        'playlist_count': 1,
        'info_dict': {
```

bashonly commented:
LGTM. In the future, please don't force push, all PR branch commits with be squashed upon merge

pukkandan commented on yt_dlp/extractor/youtube.py:
```diff
@@ -6965,22 +6965,29 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
     IE_NAME = 'youtube:search'
     _SEARCH_KEY = 'ytsearch'
-    _SEARCH_PARAMS = 'EgIQAQ%3D%3D'  # Videos only
+    _SEARCH_PARAMS = 'EgIIAfABAQ=='  # Videos only
```
Are you sure this is correct? The previous params were % encoded

bashonly replied:
Tested and works. They are encoded in the query anyways

bashonly commented on yt_dlp/extractor/youtube.py:
```diff
@@ -6965,22 +6965,30 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
+    _SEARCH_PARAMS = 'EgIIAfABAQ=='  # Videos only
     _TESTS = [{
         'url': 'ytsearch5:youtube-dl test video',
         'playlist_count': 5,
```
the new params yield only 2 results, while the old params still yield 5

pukkandan replied:
my bad. Seems I copied the wrong params. What we want is https://github.com/iv-org/invidious/blame/99a5e9cbc44daa4555f36b43bc4b7246aee454c9/spec/invidious/search/yt_filters_spec.cr#L23

pukkandan commented on yt_dlp/extractor/youtube.py:
```diff
@@ -6965,22 +6965,30 @@ class YoutubeSearchIE(YoutubeTabBaseInfoExtractor, SearchInfoExtractor):
     IE_NAME = 'youtube:search'
     _SEARCH_KEY = 'ytsearch'
-    _SEARCH_PARAMS = 'EgIQAQ%3D%3D'  # Videos only
+    _SEARCH_PARAMS = 'EgIIAfABAQ=='  # Videos only
```
```suggestion
    _SEARCH_PARAMS = 'EgIQAfABAQ=='  # Videos only
```
